scipy/sparse/_base.py

```diff
@@ -656,26 +717,40 @@ def __pow__(self, other):
 
             if other == 0:
                 from ._construct import eye
-                return eye(self.shape[0], dtype=self.dtype)
```

Comment:
I'd prefer to keep using `eye` here, and in the `self._is_array` case we can wrap the result with `dia_array` before returning.

Reply:
Done.